capreolus/searcher/special.py

```diff
@@ -0,0 +1,116 @@
+import os
+from collections import defaultdict
+from tqdm import tqdm
+
+from capreolus import ConfigOption, Dependency, constants
+from capreolus.utils.loginit import get_logger
+from capreolus.utils.trec import load_trec_topics, topic_to_trectxt
+
+from . import Searcher
+from .anserini import BM25
+
+logger = get_logger(__name__)
+
+
+class MsmarcoPsgSearcherMixin:
+    @staticmethod
+    def convert_to_trec_runs(msmarco_top1k_fn, train):
+        runs = defaultdict(dict)
+        with open(msmarco_top1k_fn, "r", encoding="utf-8") as f:
+            for line in f:
+                if train:
+                    qid, pos_pid, neg_pid = line.strip().split("\t")
+                    runs[qid][pos_pid] = len(runs.get(pos_pid, []))
```

Comment:
I'm afraid I didn't understand this part. `runs[qid][pos_pid]` would always be 0 here (for training), since `runs.get(pos_pid, [])` would always return an empty list because the keys of `runs` are `qids`, not `docids`. Did I understand that correctly?

Reply:
oh yea that's totally my fault. i guess it didn't show because in reranking or training the order doesn't matter, fixed in 483393c